From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cryptite <cryptite@gmail.com>
Date: Fri, 29 Oct 2021 17:06:43 -0500
Subject: [PATCH] Slice: Very far tracking range


diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index 7740a4d7652d23460017ca2abfc3dcd838ce9e99..56271f23fc8fbaa4eccc80fa61e4901149176189 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -1994,8 +1994,15 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
         // Paper end
         if (!(entity instanceof EnderDragonPart)) {
             EntityType<?> entitytypes = entity.getType();
-            int i = entitytypes.clientTrackingRange() * 16;
-            i = org.spigotmc.TrackingRange.getEntityTrackingRange(entity, i); // Spigot
+            // Slice
+            int i;
+            if (entity.visibleFromVeryFar) {
+                i = 128;
+            } else {
+                i = entitytypes.clientTrackingRange() * 16;
+                i = org.spigotmc.TrackingRange.getEntityTrackingRange(entity, i); // Spigot
+            }
+            // Slice end
 
             if (i != 0) {
                 int j = entitytypes.updateInterval();
diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 297ecb57f253296be201001ed23113418a7c4dd1..e8d30b8af62df36dd159b01f76d75eb110e697c9 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -344,6 +344,8 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, n
     public final gg.airplane.entity.CollisionCache collisionCache = new gg.airplane.entity.CollisionCache(this);
     // Airplane end
 
+    public boolean visibleFromVeryFar; // Slice
+
     public float getBukkitYaw() {
         return this.yRot;
     }
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index 47554042c83127f1f37e9b0881ef7c7796edd35a..c275d48132f3d0cd647b6cc957c16c2bda659e7d 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1076,6 +1076,12 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         this.getHandle().setSilent(flag);
     }
 
+    // Slice start
+    public void setVisibleFromVeryFar(boolean flag) {
+        this.getHandle().visibleFromVeryFar = flag;
+    }
+    // Slice end
+
     @Override
     public boolean hasGravity() {
         return !this.getHandle().isNoGravity();
